Uvicorn worker needs to be updated for Gunicorn

Recently, I realized that the uvicorn worker class has been deprecated and needs to be updated to a particular package on PyPI.

More details here: https://www.uvicorn.org/#running-with-gunicorn

With this, we should check if the new non-deprecated worker class notifies that the worker is alive, because I found that the current one shuts down after the timeout is reached.

For example, in the ClimateDT VM, we have a timeout of 10 minutes:


```bash
58b75184fadd [2025-06-26 08:00:55 +0000] [18050] [INFO] Booting worker with pid: 18050
58b75184fadd [2025-06-26 08:00:56 +0000] [18050] [INFO] Started server process [18050]
58b75184fadd [2025-06-26 08:00:56 +0000] [18050] [INFO] Waiting for application startup.
58b75184fadd [2025-06-26 08:00:56 +0000] [18050] [INFO] Application startup complete.
58b75184fadd [2025-06-26 08:01:23 +0000] [18050] [INFO] GET /v4/auth/verify-token|RECEIVED
58b75184fadd [2025-06-26 08:01:23 +0000] [18050] [INFO] GET /v4/auth/verify-token|RTIME|0.004s
58b75184fadd [2025-06-26 08:01:23 +0000] [18050] [INFO] GET /v3/tree/t076|RECEIVED
58b75184fadd [2025-06-26 08:01:24 +0000] [18050] [WARNING] job_packages table empty, trying wrapper_job_packages
58b75184fadd [2025-06-26 08:01:24 +0000] [18050] [INFO] Process: 18050 tree workers: {}
58b75184fadd [2025-06-26 08:01:24 +0000] [18050] [INFO] GET /v3/tree/t076|RTIME|0.991s
58b75184fadd [2025-06-26 08:01:28 +0000] [18050] [INFO] GET /v3/joblog/t076_19900101_fc0_1_LRA_GENERATOR.20250619155310.err|RECEIVED
58b75184fadd [2025-06-26 08:01:28 +0000] [18050] [INFO] GET /v3/joblog/t076_19900101_fc0_1_LRA_GENERATOR.20250619155310.err|RTIME|0.005s
58b75184fadd [2025-06-26 08:04:50 +0000] [18050] [INFO] GET /v4/auth/verify-token|RECEIVED
58b75184fadd [2025-06-26 08:04:50 +0000] [18050] [INFO] GET /v4/auth/verify-token|RTIME|0.002s
58b75184fadd [2025-06-26 08:04:51 +0000] [18050] [INFO] GET /v3/tree/t07a|RECEIVED
58b75184fadd [2025-06-26 08:04:52 +0000] [18050] [WARNING] job_packages table empty, trying wrapper_job_packages
58b75184fadd [2025-06-26 08:04:52 +0000] [18050] [INFO] Process: 18050 tree workers: {}
58b75184fadd [2025-06-26 08:04:52 +0000] [18050] [INFO] GET /v3/tree/t07a|RTIME|0.840s
58b75184fadd [2025-06-26 08:04:58 +0000] [18050] [INFO] GET /v3/joblog/t07a_19900101_fc0_1_2_TRANSFER.20250626110137.err|RECEIVED
58b75184fadd [2025-06-26 08:04:58 +0000] [18050] [INFO] GET /v3/joblog/t07a_19900101_fc0_1_2_TRANSFER.20250626110137.err|RTIME|0.005s
58b75184fadd [2025-06-26 08:05:54 +0000] [18050] [INFO] GET /v3/performance/o007|RECEIVED
58b75184fadd [2025-06-26 08:10:56 +0000] [9] [CRITICAL] WORKER TIMEOUT (pid:18050)
58b75184fadd [2025-06-26 08:10:56 +0000] [9] [ERROR] Worker (pid:18050) was sent code 134!
```

Then, even if the worker processed requests, the shutdown is done exactly 10 minutes after the boot of the worker.

--- a/autosubmit_api/cli.py
+++ b/autosubmit_api/cli.py
@@ -47,7 +47,7 @@ def start_app_gunicorn(
     options = {  # Options to always have
         "preload_app": True,
         "capture_output": True,
-        "worker_class": "uvicorn.workers.UvicornWorker",
+        "worker_class": "uvicorn_worker.UvicornWorker",
         "timeout": 600, # Change the default timeout to 10 minutes
         **kwargs,
     }

--- a/setup.py
+++ b/setup.py
@@ -36,6 +36,7 @@ install_requires = [
     "SQLAlchemy~=2.0.23",
     "python-cas~=1.6.0",
     "autosubmit==4.1.11",
+    "uvicorn-worker~=0.3.0",
 ]
 
 # Test dependencies

